src/Swarm/Language/Syntax.hs

```diff
@@ -1111,3 +1112,24 @@ freeVarsV = freeVarsT . (\f -> \case TVar x -> TVar <$> f x; t -> pure t)
 -- | Apply a function to all free occurrences of a particular variable.
 mapFreeS :: Var -> (Syntax' ty -> Syntax' ty) -> Syntax' ty -> Syntax' ty
 mapFreeS x f = freeVarsS %~ (\t -> case t ^. sTerm of TVar y | y == x -> f t; _ -> t)
+
+-- | Each constructor is a assigned a value of 1, plus
+-- any recursive syntax it entails.
+--
+-- Note that in this case statement, all of the constructors
+-- with nested Syntax are given explicit
+-- cases, whereas a catchall for non-recursive
+-- constructors is placed at the end.
+-- This catch-all is "dangerous" if we ever
+-- add more recursive terms to the language; the
+-- compiler won't warn us about missing cases.
```

Comment:
I'd rather just list all the other cases explicitly.  Putting this comment here doesn't solve anything.  If my past experience is any guide, if we leave the catch-all case, we WILL add another AST node in the future, we WILL forget to update this code, and it WILL eventually bite us.

Reply:
On second thought, instead we should just do this one simple future-proof trick:
```
import Data.Lens.Plated (universe)

measureAstSize :: Data a => Term' a -> Int
measureAstSize = length . universe
```
:tada: 

`universe` generates a list of all recursive subterms of the same type, which is exactly what we want to count here.  You should check that this gives the same results as your explicit version was giving, but I think it should.

Reply:
> `measureAstSize = length . universe`

That is slick.  I've come up with something similar, pushed in the latest commit: count the nodes of the `Tree` generated using code from #1014.  Does there happen to exist any Lens magic to automatically generate this tree?

Reply:
There sure does, see my comment on #1014 .